Lib/fontTools/varLib/instancer/__init__.py

```diff
@@ -1273,9 +1271,57 @@ def instantiateVariableFont(
         },
     )
 
+    # Set Regular/Italic/Bold/Bold Italic bits as appropriate, after the name
+    # table has been updated.
+    setRibbiBits(varfont)
+
     return varfont
 
 
+def setRibbiBits(font):
+    """Set the `head.macStyle` and `OS/2.fsSelection` style bits
+    appropriately."""
+
+    english_ribbi_style = font["name"].getName(2, 3, 1, 0x409)
+    if english_ribbi_style is None:
+        log.warning(
```

Comment:
i think all these logging warnings are too noisy, make them debug or remove them. If one doesn't have those names it may just mean they are not interested in legacy style linking behavior.